dbms/src/Storages/Transaction/TiDB.h

```diff
@@ -179,6 +179,10 @@ struct ColumnInfo
 
     ColumnID id = -1;
     String name;
+
+    /// please be very careful when you have to use offset,
+    /// because we never update offset when DDL action changes.
+    /// Thus, our offset will not exactly correspond the order of columns.
     Int32 offset = -1;
```

Comment:
can we make `ColumnInfo.offset` and `IndexColumnInfo.offset` to be private to avoid them being used unexpectedly?